src/metabase/lib/fe_util.cljc

```diff
@@ -151,6 +153,15 @@
       _
       (lib.metadata.calculation/display-name query stage-number filter-clause))))
 
+(defn- query-dependents-foreign-keys
+  [metadata-providerable columns]
+  (for [column columns
+        :let [fk-target-field-id (:fk-target-field-id column)]
+        :when (and fk-target-field-id (lib.types.isa/foreign-key? column))]
+    (if-let [fk-target-field (lib.metadata/field metadata-providerable fk-target-field-id)]
```

Comment:
For implicit joins we need the whole table. If the field is available, get `table-id` from it. Otherwise we need another FE-BE round-trip to fetch missing metadata.